### PR TITLE
Update Scala 3 macro reflection

### PR DESCRIPTION
**Pull request by nicolasstucki**
This fixes https://github.com/lampepfl/scala3-macro-tutorial/issues/31 and fixes https://github.com/lampepfl/dotty/issues/10992

### CODE REVIEW

File: _overviews/scala3-macros/tutorial/reflection.md [hunk 1]
```diff
@@ -16,19 +16,19 @@ We can use `scala.quoted.quotes` to import it.
...
+The full imported API can be found here: [Reflection](https://dotty.epfl.ch/api/scala/quoted/Quotes$reflectModule.html?query=trait%20reflectModule)
 
 For example to find what is a `Term`, we can see in the hierarchy that it is a subtype of `Statement` which is a subtype of `Tree`.
-If we look into the [`TermMethods`](https://dotty.epfl.ch/api/scala/tasty/Reflection/TermMethods.html) we will find all the extension methods that are defined for `Term` such as `Term.tpe` which returns a `Type`.
-As it is a subtype of `Tree` we can also look into the [`TreeMethods`](http://dotty.epfl.ch/api/scala/tasty/Reflection/TreeMethods.html) to find more methods such as `Tree.pos`.
-Each type also a module with some _static-ish_ methods, for example in the [TypeModule](http://dotty.epfl.ch/api/scala/tasty/Reflection/TypeModule.html) we can find the method `Type.of[T]` with will create an instance of `Type` containing `T`.
+If we look into the [`TermMethods`](https://dotty.epfl.ch/api/scala/quoted/Quotes$reflectModule$TermMethods.html) we will find all the extension methods that are defined for `Term` such as `Term.tpe` which returns a `Type`.
```
Comment (b-studios) [inline _overviews/scala3-macros/tutorial/reflection.md:29]:
```suggestion
If we look into [`TermMethods`](https://dotty.epfl.ch/api/scala/quoted/Quotes$reflectModule$TermMethods.html) we will find all the extension methods that are defined for `Term` such as `Term.tpe` which returns a `Type`.
```

File: _overviews/scala3-macros/tutorial/reflection.md [hunk 2]
```diff
@@ -16,19 +16,19 @@ We can use `scala.quoted.quotes` to import it.
...
+As it is a subtype of `Tree` we can also look into the [`TreeMethods`](https://dotty.epfl.ch/api/scala/quoted/Quotes$reflectModule$TreeMethods.html) to find more methods such as `Tree.pos`.
```
Comment (b-studios) [inline _overviews/scala3-macros/tutorial/reflection.md:30]:
```suggestion
As it is a subtype of `Tree` we can also look into [`TreeMethods`](https://dotty.epfl.ch/api/scala/quoted/Quotes$reflectModule$TreeMethods.html) to find more methods such as `Tree.pos`.
```

File: _overviews/scala3-macros/tutorial/reflection.md [hunk 3]
```diff
@@ -16,19 +16,19 @@ We can use `scala.quoted.quotes` to import it.
...
+Each type also a module with some _static-ish_ methods, for example in the [`TypeReprModule`](https://dotty.epfl.ch/api/scala/quoted/Quotes$reflectModule$TypeReprModule.html) we can find the method `TypeRepr.of[T]` with will create an instance of `Type` containing `T`.
```
Comment (b-studios) [inline _overviews/scala3-macros/tutorial/reflection.md:31]:
```suggestion
Each type also includes a module with some _static-ish_ methods, for example in the [`TypeReprModule`](https://dotty.epfl.ch/api/scala/quoted/Quotes$reflectModule$TypeReprModule.html) we can find the method `TypeRepr.of[T]` with will create an instance of `Type` containing `T`.
```

**Comment by b-studios** [comment]
Are all links fixed? If yes, then maybe you can also remove the workaround in:

https://github.com/scala/docs.scala-lang/blob/master/scripts/ci.sh#L19
```diff
- --url-ignore '/https://github.com/scala/docs.scala-lang/blob/master/.*/,/www.oracle.com/,/.*dotty.epfl.ch\/api\/scala\/tasty\/Reflection.*/'
+ --url-ignore '/https://github.com/scala/docs.scala-lang/blob/master/.*/,/www.oracle.com/'
```

**Comment by nicolasstucki** [comment]
Updted `scripts/ci.sh`

**Pull request merged**